firmware/thruna/PDM/src/app/states/app_driveState.c

```diff
@@ -0,0 +1,49 @@
+#include "states/app_driveState.h"
+#include "states/app_allStates.h"
+#include "App_SharedMacros.h"
+#include "App_CanUtils.h"
+#include "app_powerManager.h"
+#include "App_CanTx.h"
+#include "App_CanRx.h"
+#include "states/app_initState.h"
+
+static void driveStateRunOnEntry(struct StateMachine *const state_machine)
+{
+    App_CanTx_PDM_State_Set(PDM_DRIVE_STATE);
+    app_powerManager_setState(POWER_MANAGER_DRIVE);
+}
+
+static void driveStateRunOnTick1Hz(struct StateMachine *const state_machine)
+{
+    app_allStates_runOnTick1Hz(state_machine);
+}
+
+static void driveStateRunOnTick100Hz(struct StateMachine *const state_machine)
+{
+    app_allStates_runOnTick100Hz(state_machine);
+    bool isCorrectState =
+        App_CanRx_BMS_State_Get() != BMS_DRIVE_STATE || App_CanRx_BMS_State_Get() != BMS_INVERTER_ON_STATE;
+
+    if (isCorrectState)
+    {
+        App_SharedStateMachine_SetNextState(state_machine, app_initState_get());
```

Comment:
I thought the behavior we wanted was we just stay in drive state indefinitely? 
I don't think we took notes on this discussion, but we probably should have, my bad. How are you implementing it here?